viper/src/verifier.rs

```diff
@@ -257,9 +259,13 @@ impl<'a> Verifier<'a> {
                             .unwrap_result(verification_error_wrapper.call_fullId(viper_error)),
                     );
 
+                    let offending_node = self
+                        .jni
+                        .unwrap_result(verification_error_wrapper.call_offendingNode(viper_error));
+
                     let pos = self
                         .jni
-                        .unwrap_result(verification_error_wrapper.call_pos(viper_error));
+                        .unwrap_result(error_node_positioned_wrapper.call_pos(offending_node));
```

Comment:
We could also store the position id of the offending node in a new `offending_pos_id` field of `VerificationError`.

Reply:
In Silver itself you mean?

Reply:
I mean here in Prusti, the `viper` crate. Instead of storing
```
VerificationError::pos_id <-- error.offendingNode.pos.id
```
we could store
```
VerificationError::pos_id <-- error.pos.id // as before
VerificationError::offending_pos_id <-- error.offendingNode.pos.id // new
```
The advantage is that the field name would reflect better from where we got the position id. Just in case Viper changes again and we have to debug what's going on.